doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    DuckDB \cite{duckdb}, Clickhouse \cite{clickhouse}, and Snowflake \cite{snowflake} all do.
+
+    Regardless of whether an engine has explicit support for as-of joins, there
+    are complexities with putting SQL in front of the user and telling them to
+    write complex N-way as-of joins. First, while the robotics developer may have
+    some basic SQL experience, they are not a SQL expert, and particularly if the
+    dialect does not support explicit as-of joins the queries can become
+    extremely complex. Although one workable option is to hire SQL experts to
+    interface between the database and the robotics developer, one of dp3's
+    experimental goals is to avoid this solution and attempt instead to empower
+    the robotics developer directly. Second, analytics databases are not
+    general-purpose databases and often have non-obvious performance footguns and
+    tricks that you need to understand to really use them effectively. A typical
+    user with basic SQL experience has usually gained that experience from a
+    general purpose RDBMS, and may be aggravated to find that the SQL
+    implementation of an analytics DBMS comes with a host of unwritten guidance
+    along the lines of \q{don't use joins} or \q{manually push down where
+    clauses}, or that join algorithms must be directly specified. SQL knowledge
+    is portable to a degree, but perhaps less than might initially appear.
+
+    In dp3 we try to reduce the surface for confusion with a more concise and
+    narrowly-scoped query language specifically targeted at expressing as-of
+    joins. While we compile these queries to our own execution logic, the binding
+    of the language to dp3 is not very tight, and there would be no conceptual
+    barrier to instead compiling to e.g Clickhouse SQL in a way that
+    transparently handles the quirks of Clickhouse for the user and results in a
+    statement that should execute efficiently. This could be a useful extension
+    strategy for a platform built on dp3.
+
+    \section{Implementation}
+    \subsection{Program architecture}
+    \begin{figure}
+      \begin{forest}
+        for tree={
+          draw,
+          semithick,
+          minimum height = 3ex,
+          minimum width = 2em,
+          grow = south,
+          forked edge,
+          s sep = 3mm,
+          l sep = 6mm,
+          fork sep = 3mm,
+        }
+        [router
+        [treemgr
+        [rootmap
+        [sql]]
+        [versionstore
+        [sql]]
+        [walmgr
+        [disk]]
+        [nodestore
+        [cache]
+        [S3]]]]
+      \end{forest}
+      \caption{Component structure of dp3}
+      \label{modstructure}
+    \end{figure}
+
+    The component structure of dp3 is depicted in \autoref{modstructure}. This
+    section will briefly outline the main data flows, and the following sections
+    will explain components in greater detail.
+
+    \subsubsection{Write (import) path}
```

Comment:
I don't like import in here, I think that was from some previous feedback. Want to remove it now.